src/main/java/net/openhft/chronicle/core/io/DualReferenceCounted.java

```diff
@@ -49,6 +49,18 @@ public StackTrace createdHere() {
         return a.createdHere();
     }
 
+    @Override
+    public void addReferenceChangeListener(ReferenceChangeListener referenceChangeListener) {
+        a.addReferenceChangeListener(referenceChangeListener);
+        b.addReferenceChangeListener(referenceChangeListener);
+    }
+
+    @Override
+    public void removeReferenceChangeListener(ReferenceChangeListener referenceChangeListener) {
+        a.removeReferenceChangeListener(referenceChangeListener);
+        b.removeReferenceChangeListener(referenceChangeListener);
```

Comment:
This is nasty, if one day we had two ReferenceCounted's that both supported listeners. I'm not sure what to do here. The pattern thus far seems to be do things to `a` that make sense to only happen once, but in this case, currently, `a` is the TracingReferenceCounted. So I'd have to re-implement the listener handling in there and you'd get a different implementation when reference tracing was turned on or off.
Perhaps DualReferenceCounted should have explicitly typed `a` and `b` so we can bake assumptions in. I don't think being generic is very helpful here. Perhaps it should also be internal?

Reply:
DualReferenceCounted is now deprecated